lms/services/canvas_api/factory.py

```diff
@@ -12,16 +12,20 @@ def canvas_api_client_factory(_context, request):
     :param request: Pyramid request object
     :return: An instance of CanvasAPIClient
     """
-    ai_getter = request.find_service(name="ai_getter")
+    application_instance = request.find_service(name="application_instance").get()
```

Comment:
So just _constructing_ the Canvas API client can raise `ConsumerKeyError`. Which was already the case before. It's not actually a problem because nothing tries to use this service without a valid consumer key